<!DOCTYPE html>
<html lang="en">
	<body>
		<script>
			// Making AJAX Calls with Fetch and Promises
			function getWeather(place) {
				fetch(
					`https://crossorigin.me/https://www.metaweather.com/api/location/${place}/`
				)
					.then((response) => {
						console.log(response);
						return response.json();
					})
					.then((data) => {
						console.log(data);
						const today = data.consolidated_weather[0];
						console.log(
							`Min Temp: ${today.min_temp} & Max Temp: ${today.max_temp}`
						);
					})
					.catch((error) => console.log(error));
			}

			getWeather(2487956);

			// Making AJAX Calls with Fetch with Async/Await
			async function getWeatherAW(place) {
				try {
					const result = await fetch(
						`https://crossorigin.me/https://www.metaweather.com/api/location/${place}/`
					);
					const data = await result.json();
					const today = data.consolidated_weather[0];
					console.log(
						`Min Temp: ${today.min_temp} & Max Temp: ${today.max_temp}`
					);

					return data;
				} catch (error) {
					console.log(error);
				}
			}

			getWeatherAW(2487956);

			let dataLondon;
			getWeatherAW(44418).then((data) => {
				dataLondon = data;
				console.log(dataLondon);
			});

			/* ************************************************************** */

			fetch("https://api.randomuser.me/?nat=US&results=1")
				.then((res) => res.json())
				.then((json) => json.results)
				.then(console.log)
				.catch(console.error);

			// Another popular approach for handling promises is to create an async function. Some developers prefer the syntax of
			// async functions because it looks more familiar, like code that’s found in a synchronous function. Instead of waiting
			// for the results of a promise to resolve and handling it with a chain of then functions, async functions can be told
			// to wait for the promise to resolve before further executing any code found in the function.
			var getFakePerson = async () => {
				let res = await fetch("https://api.randomuser.me/?nat=US&results=1");
				console.log(res);
				let { results } = res.json();
				console.log(results);
			};
			getFakePerson();

			// When using async and await, you need to surround your promise call in a try…catch block to handle any errors that
			// may occur due to an unresolved promise.
			var getFakePerson = async () => {
				try {
					let res = await fetch("https://api.randomuser.me/?nat=US&results=1");
					console.log(res);
					let { results } = res.json();
					console.log(results);
				} catch (error) {
					console.error(error);
				}
			};
			getFakePerson();
		</script>
	</body>
</html>
